src/SFML/Window/Win32/WindowImplWin32.cpp

```diff
@@ -460,6 +460,22 @@ bool WindowImplWin32::hasFocus() const
 }
 
 
+////////////////////////////////////////////////////////////
+void WindowImplWin32::setIMEPreeditPosition(const Vector2i& position)
+{
+    HIMC himc;
+    if ((himc = ImmGetContext(m_handle)) != nullptr)
+    {
+        COMPOSITIONFORM cf;
+        cf.ptCurrentPos.x = static_cast<LONG>(position.x);
+        cf.ptCurrentPos.y = static_cast<LONG>(position.y);
+        cf.dwStyle        = CFS_FORCE_POSITION; // Don't let the IME adjust the position
+        ImmSetCompositionWindow(himc, &cf);
+        ImmReleaseContext(m_handle, himc);
+    }
```

Comment:
```suggestion
    if (HIMC himc = ImmGetContext(m_handle); himc != nullptr)
    {
        COMPOSITIONFORM cf;
        cf.ptCurrentPos.x = static_cast<LONG>(position.x);
        cf.ptCurrentPos.y = static_cast<LONG>(position.y);
        cf.dwStyle        = CFS_FORCE_POSITION; // Don't let the IME adjust the position
        ImmSetCompositionWindow(himc, &cf);
        ImmReleaseContext(m_handle, himc);
    }
```

Reply:
Ah yes, my coding habits are not quite C++17 yet :laughing: